src/toil/job.py

```diff
@@ -60,13 +61,38 @@ def __init__(self, memory=None, cores=None, disk=None, cache=None, checkpoint=Fa
         exhausting all their retries, remove any successor jobs and rerun this job to restart the subtree. \
         Job must be a leaf vertex in the job graph when initially defined, \
         see :func:`toil.job.Job.checkNewCheckpointsAreCutVertices`.
-        :type cores: int or string convertable by bd2k.util.humanize.human2bytes to an int
-        :type disk: int or string convertable by bd2k.util.humanize.human2bytes to an int
-        :type cache: int or string convertable by bd2k.util.humanize.human2bytes to an int
-        :type memory: int or string convertable by bd2k.util.humanize.human2bytes to an int
-        """
-        self.cores = cores
+        :type cores: int; Can also accept a PromisedRequirement object
```

Comment:
Read PEP 0484 for a specification . The PyCharm support web site has a page about type hinting, too.